File: dentconnect_app/lib/features/auth/models/auth_response.dart
class AuthResponse {
  final String userId;
  final String? email;
  final String? displayName;
  final String role;
  final String accessToken;
  final String refreshToken;

  const AuthResponse({
    required this.userId,
    this.email,
    this.displayName,
    required this.role,
    required this.accessToken,
    required this.refreshToken,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) => AuthResponse(
        userId: json['userId'] as String,
        email: json['email'] as String?,
        displayName: json['displayName'] as String?,
        role: json['role'] as String,
        accessToken: json['accessToken'] as String,
        refreshToken: json['refreshToken'] as String,
      );

  @override
  String toString() =>
      'AuthResponse(userId: $userId, email: $email, role: $role)';
}

File: backend/src/main/java/com/dentconnect/auth/dto/AuthResponse.java
package com.dentconnect.auth.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AuthResponse {
    private String userId;
    private String email;
    private String displayName;
    private String role;
    private String accessToken;
    private String refreshToken;
    private boolean newUser;
}
